MiscVBAFunctions/Modules/MiscTables.bas

```diff
@@ -191,3 +191,42 @@ Function TableColumnToArray(TableDicts As Collection, ColumnName As String) As V
     TableColumnToArray = arr
 End Function
 
+
+Public Sub CopyTable(InputTableName As String _
+                    , StartRange As Range _
+                    , Optional OutputTableName As String _
+                    , Optional InputWB As Workbook)
+    ' Copy a Table to the desired location. This can be in the same Workbook,
+    ' or a different workbook. The of the output Table NumberFormat
+    ' is the same as the input table's.
+    '
+    ' Args:
+    '   InputTableName: Table name that will be copied.
+    '   StartRange: Range object of the output table's destination
+    '   OutputTableName: Name of the output table.
+    '                    If left empty, the input table's name gets used.
+    '   InputWB: WorkBook of the input Table. ThisWorkBook is used if left empty.
+
+    Dim col1 As Collection
+    Dim InputLO As ListObject
+    Dim OutputLO As ListObject
+    Dim InputLOBodyRange As Range
+    Dim OutputLOBodyRange As Range
+    Dim I As Long
+    
+    If OutputTableName = "" Then
+        OutputTableName = InputTableName
+    End If
+    If InputWB Is Nothing Then Set InputWB = ThisWorkbook
+
+    Set col1 = TableToDicts(InputTableName, InputWB)
+    Set InputLO = GetLO(InputTableName, InputWB)
```

Comment:
The input table can either be a ListObject or a named range in the workbook. See `TableRange` function which is also used by `TableToDicts`